system/lib/wasmfs/syscalls.cpp

```diff
@@ -462,6 +462,32 @@ int __syscall_openat(int dirfd, intptr_t path, int flags, ...) {
   return doOpen(path::parseParent((char*)path, dirfd), flags, mode);
 }
 
+int __syscall_mknodat(int dirfd, intptr_t path, int mode, int dev) {
+  assert(dev == 0); // TODO: support special devices
+  if (mode & S_IFDIR) {
+    return -EINVAL;
+  }
+  if (mode & S_IFIFO) {
+    return -EPERM;
+  }
+  auto flags = O_CREAT;
+  auto read = bool(mode & WASMFS_PERM_READ);
+  auto write = bool(mode & WASMFS_PERM_WRITE);
+  if (read && !write) {
+    flags |= O_RDONLY;
+  } else if (write && !read) {
+    flags |= O_WRONLY;
+  } else if (read && write) {
+    flags |= O_RDWR;
+  }
```

Comment:
Adding flags to determine whether to open the file for reading or writing, etc. is not useful because we aren't actually opening the file and returning a file descriptor.
```suggestion
```